projects/yieldhubfinance/index.js

```diff
@@ -0,0 +1,27 @@
+const utils = require('../helper/utils');
+const { toUSDTBalances } = require('../helper/balances');
+
+function fetch() {
+  return async () => {
+      const response = await utils.fetchURL('https://api.yieldhub.finance/tvl');
+      let tvl = 0;
+      for (chainId in response.data) {
+        const chain = response.data[chainId];
+        for (vault in chain) {
+          tvl += chain[vault];
+        } 
+      }
+      return toUSDTBalances(tvl);
+    }
+}
+
+const chains = {
+  telos:40
+}
+
+module.exports = {
+  timetravel: false,
+  telos: {
+      tvl: fetch()
```

Comment:
Hi, thanks for the PR, but can this be calculated using on-chain calls instead? We've got a guide on how to use our SDK here: https://docs.llama.fi/list-your-project/submit-a-project

Reply:
if you are using masterchef contract, you could use this as template: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/fatfire/index.js

Reply:
@g1nt0ki - It would be difficult to calculate using on-chain calls because YieldHub is a yield aggregator with many different vaults and strategies against multiple platforms. in order to calculate the balance on a certain masterchef contract, we would need to get the balance of every strategy + vault against every platform that we currently support, meaning multiple masterchef contracts and then multiple strategies for each contract. Also, since we plan to add more vaults and strategies for every new platform or LP farm that is added on the network, we would need to continually add those contract addresses in order to use on-chain calls

Reply:
Our current API that we're calling has a config list of all the LP pools and vaults + their contracts, so when we add more vaults to the code, the API will automatically pull those TVL as well. In order to do the same in defillama, we'd have to update every time we add a new vault or platform, which wouldn't be ideal